spanner_orm/tests/model_api_test.py

```diff
@@ -49,3 +50,53 @@ def test_find_result(self, find):
     self.assertEqual(result.key, 'key')
     self.assertEqual(result.value_1, 'value_1')
     self.assertIsNone(result.value_2)
+
+  @mock.patch('spanner_orm.api.SpannerApi.insert')
+  def test_create_calls_api(self, insert):
+    mock_transaction = mock.Mock()
+    models.SmallTestModel.create(mock_transaction, key='key', value_1='value')
+
+    insert.assert_called_once()
+    (transaction, table, columns, values), _ = insert.call_args
+    self.assertEqual(transaction, mock_transaction)
+    self.assertEqual(table, models.SmallTestModel.table)
+    self.assertEqual(list(columns), ['key', 'value_1'])
+    self.assertEqual(list(values), [['key', 'value']])
+
+  def test_create_error_on_invalid_keys(self):
+    with self.assertRaises(error.SpannerError):
+      models.SmallTestModel.create(key_2='key')
+
+  def assert_api_called(self, mock_api, mock_transaction):
+    mock_api.assert_called_once()
+    (transaction, table, columns, values), _ = mock_api.call_args
+    self.assertEqual(transaction, mock_transaction)
+    self.assertEqual(table, models.SmallTestModel.table)
+    self.assertEqual(list(columns), ['key', 'value_1', 'value_2'])
+    self.assertEqual(list(values), [['key', 'value', None]])
+
+  @mock.patch('spanner_orm.api.SpannerApi.insert')
+  def test_save_batch_inserts(self, insert):
```

Comment:
this method and `test_create_calls_api` seem to have same model, yet former asserts with `value_2` column, latter doesn't. I don't see `value_2` anywhere in this method, how does it assert that column?

Reply:
batch_save() always sets all columns on a model, create() only sets columns that were explicitly passed in